### PR TITLE
[14.0][IMP] stock_analytic: Add analytic on move lines tree view + store analytic account

### PR DESCRIPTION
**Pull request by rousseldenis**
This was intended to show the analytic account on move lines level.

So:

- [x] Add the analytic account on move line level views
- [x] Store the analytic account on move lines level in order to support extra move behaviour

### CODE REVIEW

**Review by AaronHForgeFlow — APPROVED** [review]
Code review LGTM

**Review by Mantux11 — APPROVED** [review]

**Comment by AaronHForgeFlow** [comment]
It seems there are conflicts in the branch :S

**Review by adrienpeiffer — CHANGES_REQUESTED** [review]
LGTM

**Review by adrienpeiffer — APPROVED** [review]
LGTM

File: stock_analytic/migrations/14.0.2.0.0/pre-migrate.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+            "stock_move_line",
+            "many2one",
+            "int4",
+            "stock_analytic",
+        )
+    ]
+    openupgrade.add_fields(env, analytic_field)
+    query = """
+        UPDATE stock_move_line
+            SET analytic_account_id = sm.analytic_account_id
```
Comment (adrienpeiffer) [inline stock_analytic/migrations/14.0.2.0.0/pre-migrate.py:20]: Finally, one question. Is the analytic_account_id existing before update the module ?

Comment (adrienpeiffer) [reply stock_analytic/migrations/14.0.2.0.0/pre-migrate.py:20]: @rousseldenis Event it's merged, could you clarify ?

Comment (rousseldenis) [reply stock_analytic/migrations/14.0.2.0.0/pre-migrate.py:20]: Yes, field is created before pre-migrate.

Comment (rousseldenis) [reply stock_analytic/migrations/14.0.2.0.0/pre-migrate.py:20]: @adrienpeiffer You want to say, maybe a condition is missing.

```suggestion
            SET analytic_account_id = sm.analytic_account_id
            WHERE analytic_account_id IS NULL
```

**Pull request merged**